XenAdmin/Commands/Controls/ContextMenuBuilder.cs

```diff
@@ -877,8 +879,9 @@ public override void Build(IMainWindow mainWindow, SelectedItemCollection select
                 items.AddIfEnabled(new VappShutDownCommand(mainWindow, selection));
                 items.AddSeparator();
 
-                items.AddIfEnabled(new StartVMOnHostToolStripMenuItem(mainWindow, selection, true));
-                items.AddIfEnabled(new ResumeVMOnHostToolStripMenuItem(mainWindow, selection, true));
+                // Hide the menu if pool size is too large.
+                items.AddIf(new StartVMOnHostToolStripMenuItem(mainWindow, selection, true), () => { return hostsCount <= MAX_HOSTS_IN_CONTEXT_MENU; });

```

Comment:
Wouldn't this be simpler if it was just:
```
if(hostsCount <= MAX_HOSTS_IN_CONTEXT_MENU){
    items.AddIfEnabled(new StartVMOnHostToolStripMenuItem(mainWindow, selection, true));
    items.AddIfEnabled(new ResumeVMOnHostToolStripMenuItem(mainWindow, selection, true));
}
```

IMO it's confusing to have `addIf` which checks the condition **and** enabled, and we don't really gain anything by using a lambda wrapper here.